src/rotate.jl

```diff
@@ -0,0 +1,102 @@
+
+#
+# definition for rotating points
+# 
+
+"""
+Rotate the point X using the rotation parameters specified in R (i.e. R * X)
+
+rotate_point(R::T, X)
+
+"""
+rotate_point{T <: RotationTypes, U <: Real}(R::T, X::Vec{3,U})  =  rotate_point(RotMatrix(R), X)    # generic version, go through a rotation matrix
```

Comment:
OK, I'm happy with this instead of `*`. People can make a matrix if that's what they want!

Can I suggest `rotate()` as the function name? People tend to like verb-functions.